### PR TITLE
Decomposite the reporter 

### PR DESCRIPTION
**Pull request by okozlovlux**
Decomposited the reporter (xmlReporter, atf_logger, sdl_logger)
 - logging functions moved to the atf_logger and sdl_logger 
 - creation of the atf_logger moved to the util.lua
 - creation of the sdl_logger moved to the SDL.lua

Related to APPLINK-16791

@LuxoftAKutsan @anosach-luxoft @LevchenkoS @Kozoriz @VVeremjova @vlantonov @pvvasilev please review

### CODE REVIEW

**Comment by LevchenkoS** [comment]
@okozlovlux, reviewed.

**Comment by pvvasilev** [comment]
Reviewed.

**Comment by vlantonov** [comment]
@okozlovlux Reviewed.

File: modules/atf_logger.lua [hunk 1]
```diff
@@ -23,24 +46,91 @@ function Logger:SDLtoMOB(message)
...
+
+function Logger.LOGTestCaseStart(test_case)
+  if config.excludeReport then return end
+  Logger:StartTestCase(test_case)
+end
+
+function Logger.LOG(tract, message)
+  if config.excludeReport then return end
+  if config.storeFullATFLogs then    
+    Logger[tract](Logger, message)
```
Comment (VVeremjova) [inline modules/atf_logger.lua]: @okozlovlux If storeFullATFLogs is true and message from HMI then it should be written twice?

File: modules/mobile_connection.lua [hunk 2]
```diff
@@ -44,7 +46,7 @@ function module.mt.__index:OnInputData(func)
     local msg = protocol_handler:Parse(binary)
     for _, v in ipairs(msg) do
       -- After refactoring should be moved in mobile session
-      xmlReporter:LOG("SDLtoMOB", v)
+      atf_logger.LOG("SDLtoMOB", v)
```
Comment (VVeremjova) [inline modules/mobile_connection.lua]: @okozlovlux You replaced xml reporter on atf logger. Are you sure that this information should not also be kept in xml file?

File: modules/atf_logger.lua (hunk 1 above)
Comment (okozlovlux) [inline modules/atf_logger.lua]: @VVeremjova , yes. but it writes in different files

File: modules/SDL.lua [hunk 3]
```diff
@@ -1,14 +1,15 @@
...
 
 SDL.exitOnCrash = true
 SDL.STOPPED = 0
 SDL.RUNNING = 1
 SDL.CRASH = -1
 
-function SDL:StartSDL(pathToSDL, smartDeviceLinkCore, ExitOnCrash)
-  xmlReporter:initSDLLOG()
+function SDL:StartSDL(pathToSDL, smartDeviceLinkCore, ExitOnCrash)  
+  sdl_logger.init_log("SLD.lua")
```
Comment (VVeremjova) [inline modules/SDL.lua]: @okozlovlux  Why SLD.lua?

Comment (okozlovlux) [inline modules/SDL.lua]: @VVeremjova , typo. I will change it to `SDL.lua`

File: modules/websocket_connection.lua [hunk 4]
```diff
@@ -1,7 +1,11 @@
 local json = require("json")
-local module = { mt = { __index = {} } }
+
+local module = { 
+  mt = { __index = {} }
+}
 
 function module.WebSocketConnection(url, port)
+  --atf_logger.init_log("MyScriptName")
```
Comment (VVeremjova) [inline modules/websocket_connection.lua]: @okozlovlux Commented code

File: modules/mobile_connection.lua (hunk 2 above)
Comment (okozlovlux) [inline modules/mobile_connection.lua]: @VVeremjova , LOG func doesn't write to the xml file. It writes to the ATFLogs.

File: modules/atf_logger.lua (hunk 1 above)
Comment (okozlovlux) [inline modules/atf_logger.lua]: @VVeremjova , fixed in b8e5389

File: modules/SDL.lua (hunk 3 above)
Comment (okozlovlux) [inline modules/SDL.lua]: @VVeremjova , fixed in b8e5389

File: modules/websocket_connection.lua (hunk 4 above)
Comment (okozlovlux) [inline modules/websocket_connection.lua]: @VVeremjova , fixed in b8e5389

**Comment by Kozoriz** [comment]
Reviewed

**Comment by VVeremjova** [comment]
Reviewed

**Comment by anosach-luxoft** [comment]
Reviewed

File: modules/atf_logger.lua [hunk 5]
```diff
@@ -23,24 +46,91 @@ function Logger:SDLtoMOB(message)
...
 
 function Logger:SDLtoHMI(message)
   local log_str = string.format(Logger.hmi_log_format, "SDL->HMI", formated_time(), message)
   self.atf_log_file:write(log_str)
+  if config.storeFullATFLogs then
+    self.full_atf_log_file:write(log_str)
+  end
+end
+
+local function get_script_name(str)
```
Comment (LuxoftAKutsan) [inline modules/atf_logger.lua]: @okozlovlux GET SCRIPTA NAME FROM WHAT? pLEAS EADD A COMMENT

Comment (okozlovlux) [inline modules/atf_logger.lua]: @LuxoftAKutsan , fixed in 	ba3aa24

**Comment by LuxoftAKutsan** [comment]
Reviewed

**Pull request merged**